src/cpp/src/group_beam_searcher.cpp

```diff
@@ -462,9 +460,14 @@ EncodedResults beam_search(ov::InferRequest& lm,
         ) {
             results.scores.push_back(beam->get().score);
             results.tokens.push_back(std::move(beam->get().tokens));
+            selected_beams.push_back(beam->get().global_beam_idx);
         }
     }
-    return results;
+    std::optional<int32_t> selected_beam = std::nullopt;
+    if (batch_size == 1)
+        selected_beam = selected_beams[0];
```

Comment:
done

Reply:
Why did you skip
> And here you decide whether `selected_beam` or `std::nullopt` should be returned.

part? I thought it's needed to trigger `std::fill_n(beam_data, batch_size, 0);` appropriately.

Reply:
`to trigger std::fill_n(beam_data, batch_size, 0)` input argument is set on the very first generate call. At that poing we know for sure there are no selected beams and value is not set.

I just realized that in return `std::optional` is not necessary, we can always return beam idx for the highest score. If we are not in the chat\don't need to continue we just don't use that value no need to make it optional. `std::fill_n(be..` still will work fine, since on the very first call we set input arg `selected_beam_idx` to `nullopt`.